test/test_model.py

```diff
@@ -175,6 +178,23 @@ def test_config(self):
             # Confirm that we cannot modify config values.
             self.model.config['foo'] = 'bar'
 
+    def test_resources(self):
+        os.environ['JUJU_UNIT_NAME'] = 'test/0'
+        self.addCleanup(os.environ.pop, 'JUJU_UNIT_NAME')
+        backend = op.model.ModelBackend()
+        meta = op.charm.CharmMeta()
+        meta.resources = {'foo': None}
+        model = op.model.Model('myapp/0', meta, backend)
+
+        with self.assertRaises(RuntimeError):
+            model.resources.fetch('bar')
+
+        fake_script(self, 'resource-get', 'exit 1')
+        self.assertIsNone(model.resources.fetch('foo'))
+
+        fake_script(self, 'resource-get', 'echo /var/lib/juju/agents/unit-test-0/resources/foo/foo.tgz')
+        self.assertEqual(model.resources.fetch('foo').name, 'foo.tgz')
+
```

Comment:
We should add a test that uses 2 resources, so we can make sure that both of them return different content.

Something like:
fake_script(self, 'resource-get', 'echo /var/lib/juju/agents/unit-test-0/resources/$1/$1.tgz')

?